DllImportGenerator/DllImportGenerator/ContiguousCollectionElementMarshallingCodeContext.cs

```diff
@@ -46,22 +45,41 @@ internal sealed class ContiguousCollectionElementMarshallingCodeContext : StubCo
         /// <returns>Managed and native identifiers</returns>
         public override (string managed, string native) GetIdentifiers(TypePositionInfo info)
         {
-            var (_, native) = parentContext.GetIdentifiers(info);
+            var (_, native) = ParentContext!.GetIdentifiers(info);
             return (
-                $"{native}.ManagedValues[{indexerIdentifier}]",
-                $"{nativeSpanIdentifier}[{indexerIdentifier}]"
+                $"{native}.ManagedValues[{IndexerIdentifier}]",
+                $"{nativeSpanIdentifier}[{IndexerIdentifier}]"
             );
         }
 
         public override string GetAdditionalIdentifier(TypePositionInfo info, string name)
         {
-            return $"{nativeSpanIdentifier}__{indexerIdentifier}__{name}";
+            return $"{nativeSpanIdentifier}__{IndexerIdentifier}__{name}";
         }
 
         public override TypePositionInfo? GetTypePositionInfoForManagedIndex(int index)
         {
             // We don't have parameters to look at when we're in the middle of marshalling an array.
             return null;
         }
+
+        private static string CalculateIndexerIdentifierBasedOnParentContext(StubCodeContext? parentContext)
+        {
+            int i = 0;
+            while (parentContext is StubCodeContext context)
+            {
+                if (context is ContiguousCollectionElementMarshallingCodeContext)
+                {
+                    i++;
+                }
+                parentContext = context.ParentContext;
+            }
+
+            // Follow a progression of indexers of the following form:
+            // __i, __j, __k, __ii, __jj, __kk, etc.
+            StringBuilder indexerIdentifier = new StringBuilder("__");
+            indexerIdentifier.Append((char)('i' + (i % 3)), 1 + (i / 3));
```

Comment:
I think I would prefer to simply append the depth - `__i0`, `__i1`, ...,  `__iN`, etc. Computing the depth from the current form appears much harder. Is there a benefit of current transform?

Reply:
I like the idea of appending the depth. That's a better idea. I'll switch it to that.